src/main/java/duke/ui/MainWindow.java

```diff
@@ -0,0 +1,104 @@
+package duke.ui;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import duke.Duke;
+import duke.commands.Command;
+import duke.commands.CommandType;
+import duke.exceptions.DukeException;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
```

Comment:
Perhaps you can separate the imports from duke and the ones from javafx? Would improve readability.